File: web/src/main/java/by/sorface/passport/web/facade/accounts/RenewPasswordFacade.java
package by.sorface.passport.web.facade.accounts;

import by.sorface.passport.web.records.tokens.ApplyNewPasswordRequest;

public interface RenewPasswordFacade {

    void forgetPassword(final String email);

    void applyNewPassword(final ApplyNewPasswordRequest request);

    void checkRenewPasswordToken(final String hash);

}

File: web/src/main/kotlin/by/sorface/passport/web/facade/accounts/RenewPasswordFacade.kt
package by.sorface.passport.web.facade.accounts

import by.sorface.passport.web.records.tokens.ApplyNewPasswordRequest

interface RenewPasswordFacade {
    fun forgetPassword(email: String)

    fun applyNewPassword(request: ApplyNewPasswordRequest)

    fun checkRenewPasswordToken(hash: String)
}
